src/com/goide/runconfig/testing/coverage/GoCoverageProgramRunner.java

```diff
@@ -32,13 +36,57 @@
 import com.intellij.execution.runners.GenericProgramRunner;
 import com.intellij.execution.runners.RunContentBuilder;
 import com.intellij.execution.ui.RunContentDescriptor;
+import com.intellij.notification.*;
 import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.vfs.VirtualFile;
 import com.intellij.util.ObjectUtils;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import javax.swing.event.HyperlinkEvent;
+
 public class GoCoverageProgramRunner extends GenericProgramRunner {
   private static final String ID = "GoCoverageProgramRunner";
+  private static final String GO_GET_BINARY_LINK = "goGetPackageCoverageLink";
+  private static final String GO_GET_BINARY_NAME = "github.com/haya14busa/goverage";
+  private static final String GO_GET_BINARY = "goverage";
+
+  private class CoveragePatcher extends GoTestRunningState.ExecutorPatcher {
+    private GoTestRunConfiguration myConfiguration;
+    private CoverageEnabledConfiguration myCoverageConfiguration;
+    private VirtualFile myPackageCoverageExecutable;
+
+    public CoveragePatcher(CoverageEnabledConfiguration coverageConfiguration, GoTestRunConfiguration configuration) {
+      super(configuration);
+      myConfiguration = configuration;
+      myCoverageConfiguration = coverageConfiguration;
+      myPackageCoverageExecutable = GoSdkUtil.findExecutableInGoPath(
+          GO_GET_BINARY,
+          myConfiguration.getProject(),
+          myConfiguration.getConfigurationModule().getModule());
+    }
+
+    private boolean hasPackageCoverageExecutable() {
+      return myPackageCoverageExecutable != null;
+    }
+
+    @Override
+    public void beforeTarget(@NotNull GoExecutor executor) {
+      if (hasPackageCoverageExecutable()) {
+        executor.withExePath(myPackageCoverageExecutable.getPath());
+        executor.withParameters("-v");
+      }
+      else {
+        super.beforeTarget(executor);
```

Comment:
so if a user doesn't have goverage then you will add GoToolParams twice, here and in `afterTarget`. It leads to this:

![image](https://cloud.githubusercontent.com/assets/140920/24861290/f6179992-1e00-11e7-83c1-7cfc0acbbf27.png)